src/bpftrace.cpp

```diff
@@ -845,18 +852,28 @@ int BPFtrace::run(Output &out, BpfBytecode bytecode)
     }
   }
 
+  int num_special_attached = 0;
+
   auto begin_probe = resources.special_probes.find("BEGIN");
   if (begin_probe != resources.special_probes.end()) {
     auto &begin_prog = bytecode_.getProgramForProbe((*begin_probe).second);
     if (::bpf_prog_test_run_opts(begin_prog.fd(), nullptr))
       return -1;
 
     LOG(V1) << "Attaching BEGIN";
+    ++num_special_attached;
+  }
+
+  // This is a little bit of a lie as we don't actually attach the END
+  // special probe until the end
```

Comment:
I don't think it's a lie. In our abstract world of attachments, the probe is indeed attached and waiting for the END event to happen.

Reply:
Ok ok - perhaps the language should be we don't actually RUN the END program until the end. Anyway, I can remove.